src/ansys/geometry/core/designer/design.py

```diff
@@ -1220,3 +1220,139 @@ def _update_design_inplace(self) -> None:
+                    print(f"Added new body: {body_info['name']} (ID: {body_info['id']})")
+                    return True  # Found and added
+
+                # Recursively search in subcomponents
+                if find_and_add_body(body_info, component.components):
+                    return True
+
+            return False  # Not found
+
+        # Function to find and update bodies within components recursively
+        def find_and_update_body(body_info, component):
+            if component == []:
+                return False
+            for body in component.bodies:
+                if body.id == body_info["id"]:
+                    update_body(body, body_info)
+                    return True  # Found and updated
+
+                # Recursively search in subcomponents
+                if find_and_update_body(body_info, component.components):
+                    return True
+
+            return False  # Not found
+
+            # Function to find and remove bodies within components recursively
+
+        def find_and_remove_body(body_info, component):
+            if component == []:
+                return False
+            for body in component.bodies:
+                if body.id == body_info["id"]:
+                    # Remove the body from the component
+                    component.bodies.remove(body)
+                    print(f"Removed body: {body_info['id']}")
+                    return True  # Found and removed
+
+                # Recursively search in subcomponents
+                if find_and_remove_body(body_info, component.components):
+                    return True
+
+            return False  # Not found
+
+        # Loop through all changed bodies from the tracker
+        for body_info in tracker_response["modified_bodies"]:
+            body_id = body_info["id"]
+            body_name = body_info["name"]
+            is_surface = body_info.get(
+                "is_surface", False
+            )  # Default to False if "is_surface" is missing
+            updated = False  # Track if a body was updated
+
+            # First, check bodies at the root level
+            for body in self.bodies:
+                if body.id == body_id:
+                    update_body(body, body_info)
+                    updated = True
+                    break
+
+            # If not found in root, search within components
+            if not updated:
+                for component in self.components:
+                    updated = find_and_update_body(body_info, component)
+
+        # Loop through all deleted bodies from the tracker
+        for body_info in tracker_response["deleted_bodies"]:
+            body_id = body_info["id"]
+            removed = False  # Track if a body was removed
+
+            # First, check bodies at the root level
+            for body in self.bodies:
+                if body.id == body_id:
+                    body.is_alive = False
+                    # self.bodies.remove(body)
```

Comment:
I think we have to activate this -- self.bodies should only contain "alive" bodies. However, the proper approach is to mark the body as "dead" by changing the bool and removing it from the list. Did you experience problems when doing this?

Reply:
is_alive seems like only a getter not setter? do we want to add a setter for it too? If we are removing it from the list, I am not sure what does it do to mark it as dead?

Reply:
Ah yes -- but as a library developer you can make use of the ``_is_alive`` protected attribute to put it to False :) users aren't expected to make use of it, but we need to use it in circumstances like this one for example.

If _is_alive is False, it won't allow a user to perform gRPC calls on that object. And avoiding kernel crashes on the server side.

So, as a summary: don't create a setter, just use the protected attribute (_is_alive) and set it to False. And then, if a user is holding a reference to that object, they will also get the update and will no longer be allowed to perform server calls on that object =)